deps.windows/patches/ducible/0002-Use-py-instead-of-python.patch

```diff
@@ -0,0 +1,93 @@
+From bc650d6c64f2f3925263cf4c5fc7587363ba4875 Mon Sep 17 00:00:00 2001
+From: Rodney <dennis@obsproject.com>
+Date: Thu, 2 Mar 2023 13:35:20 +0100
+Subject: [PATCH] Use py instead of python
+
+---
+ vs/vs2015/ducible/ducible.vcxproj | 9 ++++-----
+ vs/vs2015/pdbdump/pdbdump.vcxproj | 9 ++++-----
+ 2 files changed, 8 insertions(+), 10 deletions(-)
+
+diff --git a/vs/vs2015/ducible/ducible.vcxproj b/vs/vs2015/ducible/ducible.vcxproj
+index 99d4ef4..f49dd22 100644
+--- a/vs/vs2015/ducible/ducible.vcxproj
++++ b/vs/vs2015/ducible/ducible.vcxproj
+@@ -109,7 +108,7 @@
+       <GenerateDebugInformation>true</GenerateDebugInformation>
+     </Link>
+     <CustomBuildStep>
+-      <Command>python ..\..\..\scripts\version.py ..\..\..\src\version.h.in ..\..\..\src\version.h --version-file ..\..\..\VERSION</Command>
++      <Command>py -3 ..\..\..\scripts\version.py ..\..\..\src\version.h.in ..\..\..\src\version.h --version-file ..\..\..\VERSION</Command>

```

Comment:
Does this require specific steps to work? Will Python 3 always be available as `py3` in the path?

If we patch this, it needs to work universally and can't just be patched to work on CI.

Reply:
`py` is the "Python Launcher for Windows" and is installed by default when python is installed via their official Windows installers: https://docs.python.org/3/using/windows.html#launcher

`py -3` in this case runs the script with the latest Python 3.x version installed on the system. Likewise `py -3.10` can be used to run a specific version.

Reply:
So the "Python Launcher for Windows" becomes a Windows build dependency if reproducible builds are desired?

Reply:
obs-deps itself is not reproducible right now, and that will take more time. The main reason of including ducible in obs-deps would be to avoid having to download it from *some other source* on CI or during local builds that are meant to be identical to CI builds (i.e. using the prebuilt obs-deps).

However, due to the difficulties encountered so far I am considering shelving ducible for now and then come back to it post-30 with a fork with CMake support. Just to make everyone's life a little easier.